### PR TITLE
Improve wifi coex doc 

### PR DESCRIPTION
**Pull request by ankuns**
Related to KRKNWK-16476.
It is ready for review despite being marked as draft (to not trigger full CI just because of doc change).

The chapter `Adding Wi-Fi Coexistence support to short-range radio applications` is renamed to `Coexistence of short-range radio and other radios` . It has new structure, to fit also legacy "Bluetooth-only 1-wire coexistence" which must stay at the moment.

The Bluetooth-only 1-wire coexistence was moved from "Ecosystems integration".
Added link to coexistence in "Protocols" chapter.

### CODE REVIEW

File: doc/_utils/redirects.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 """
-Copyright (c) 2022 Nordic Semiconductor
+Copyright (c) 2022-2023 Nordic Semiconductor
```
Comment (jciupis) [inline doc/_utils/redirects.py:2]: Is this change intentional?

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 2]
```diff
@@ -1,103 +1,113 @@
...
 
-Enabling Wi-Fi Coexistence
---------------------------
+Common requirements to use coexistence based on MPSL CX API:
 
-To enable Wi-Fi Coexistence, do as follows:
+1. Make sure that Kconfig option :kconfig:option:`CONFIG_MPSL` is enabled.
+   Please note that some protocol drivers like :ref:`SoftDevice Controller <nrfxlib:softdevice_controller>` enable this option by default.
+2. Make sure that Kconfig option :kconfig:option:`CONFIG_MPSL_CX` is enabled.
+3. Make sure you meet requirements specific to selected coexistence implementation.
```
Comment (jciupis) [inline doc/nrf/app_dev/wifi_coex/index.rst:55]: This statement is a bit vague in its current shape. What do you think about adding an example to help the reader understand what kind of requirements could be specific to the selected implementation? I mean something like:
> For example in case of nRF700x, additional Kconfig option :kconfig:option:`CONFIG_MPSL_CX_NRF700X` must be enabled. This requirement is specific for nRF700x coexistence implementation. Other coexistence implementations might have different requirements or have no additional requirements at all. For further details on requirements specific to selected coexistence implementation refer to the documentation of that implementation.

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 3]
```diff
@@ -135,12 +145,41 @@ To configure the hardware for a PTA inside the nRF700x series chips:
...
+   ============  =========  =================================  ==============
+   Pin           Direction  Description                        DT property
+   ============  =========  =================================  ==============
+   REQUEST       Out        Request signal                     req-gpios
+   PRIORITY      Out        SR transaction direction TX or RX  pri-dir-gpios
+   GRANT         In         Grant signal                       grant-gpios
+   ============  =========  =================================  ==============
+
+
+Enabling nRF700x Wi-Fi coexistence
```
Comment (jciupis) [inline doc/nrf/app_dev/wifi_coex/index.rst:181]: This should be `Enabling generic three wire coexistence`

File: doc/_utils/redirects.py (hunk 1 above)
Comment (ankuns) [reply doc/_utils/redirects.py:2]: I'm still trying to figure out what to put here because licence check doesn't like also the old version.

File: doc/nrf/app_dev/wifi_coex/index.rst (hunk 2 above)
Comment (ankuns) [reply doc/nrf/app_dev/wifi_coex/index.rst:55]: All these requirements are described for given implementation, please see chapter Enabling nRF700x Wi-Fi coexistence, there is about CONFIG_MPSL_CX_NRF700X.

Comment (jciupis) [reply doc/nrf/app_dev/wifi_coex/index.rst:55]: > All these requirements are described for given implementation

Yes, but the original idea of this bullet point "Make sure you meet requirements specific to selected coexistence implementation." was to make the user aware that these generic requirements common for all implementations might not be enough. If you want to avoid duplication then maybe it's enough to add something like:

> Make sure you meet requirements specific to selected coexistence implementation. For details on requirements specific to selected coexistence implementation refer to the documentation of that implementation.

In my opinion without any additional example or information this information doesn't help much.

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 4]
```diff
@@ -1,103 +1,113 @@
 .. _ug_radio_coex:
 
-Adding Wi-Fi Coexistence support to short-range radio applications
-##################################################################
+Coexistence of short-range radio and other radios
```
Comment (ahasztag) [inline doc/nrf/app_dev/wifi_coex/index.rst:3]: Is this change based on some PMT requirements? Although surely technically correct, I wonder if there really are other use cases then using this with Wi-Fi. Even if there are I doubt we can really say they are supported, as until now we've only tested it with Wi-Fi. I'm afraid if for the sake of precision we aren't loosing clearness of the text for people using the most common use case.

However if we go with the generic "other radios" I would suggest adding "e.g. WiFi" next to each "other protocols" statement .

File: doc/nrf/app_dev/wifi_coex/index.rst (hunk 3 above)
Comment (ankuns) [reply doc/nrf/app_dev/wifi_coex/index.rst]: fixed

File: doc/nrf/app_dev/wifi_coex/index.rst (hunk 2 above)
Comment (ankuns) [reply doc/nrf/app_dev/wifi_coex/index.rst:55]: @jciupis changed a bit. Is it enough?

File: doc/nrf/app_dev/wifi_coex/index.rst (hunk 4 above)
Comment (ankuns) [reply doc/nrf/app_dev/wifi_coex/index.rst:3]: Exactly what @jciupis said. There is Bluetooth-only 1-wire coexistence, which is for LTE. The term is intentionally wider than just Wi-Fi.

Comment (ankuns) [reply doc/nrf/app_dev/wifi_coex/index.rst:3]: The term Wi-Fi or LTE is in the page content. Search will match. By the way on sdk-nrf repo level this is now the only chapter  about any coexistence. Especially moved misplaced Bluetooth-only 1-wire into this chapter.

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 5]
```diff
@@ -1,103 +1,115 @@
...
+   You can provide the node using the devicetree source file of the target board or an overlay file.
+   See :ref:`zephyr:dt-guide` for more information about the DTS data structure, and :ref:`zephyr:dt_vs_kconfig` for information about differences between DTS and Kconfig.
+5. You can add a new device binding and use it as the ``compatible`` property for the node, if the provided hardware interfaces are unsuitable.
 
-1. Set the :kconfig:option:`CONFIG_MPSL_CX` Kconfig option to ``y``.
-2. Select the specific Coexistence implementation.
-   See :ref:`ug_radio_cx_wifi_coex_impls` for details about the implementations currently supported.
-#. On the nRF5340, you must apply the settings to the Kconfig options mentioned in steps 1 and 2 also to the network core.
+.. note::
+   When using the nRF5340, steps 1 and 2 above apply only for the NET core.
```
[the remark this answers is not in the document]
Comment (ankuns) [reply doc/nrf/app_dev/wifi_coex/index.rst:67]: steps 1 and 2 are about Kconfigs (CONFIG_MPSL, CONFIG_MPSL_CX)  not about pin forwarding

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 6]
```diff
@@ -1,103 +1,115 @@
...
+1. Make sure that Kconfig option :kconfig:option:`CONFIG_MPSL` is enabled.
+   Please note that some protocol drivers like :ref:`SoftDevice Controller <nrfxlib:softdevice_controller>` enable this option by default.
+2. Make sure that Kconfig option :kconfig:option:`CONFIG_MPSL_CX` is enabled.
+3. Make sure you meet requirements specific to selected coexistence implementation.
+   These include at least selecting one of supported values of :kconfig:option:`CONFIG_MPSL_CX_CHOICE` choice.
+   For details on requirements specific to selected coexistence implementation please refer to the documentation of that implementation.
+4. Make sure a ``nrf_radio_coex`` node appropriate for selected implementation is present in the devicetree.
+   When using one of the supported implementations, you must use the ``nrf_radio_coex`` name for the node.
+   However, if you add a custom user implementation, you can also use a different name.
+   Please note that some boards supported by the |NCS| provide this node out of the box.
```
[the remark this answers is not in the document]
Comment (ankuns) [reply doc/nrf/app_dev/wifi_coex/index.rst]: added reference to nrf7002dk_nrf5340

File: doc/nrf/app_dev/wifi_coex/index.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (ankuns) [reply doc/nrf/app_dev/wifi_coex/index.rst:69]: inserted point 5.

**Review by alexstanoev-nordic — APPROVED** [review]

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 7]
```diff
@@ -1,103 +1,117 @@
...
 
-Currently, the following protocols use the Wi-Fi Coexistence support provided by MPSL:
-
-   * :ref:`ug_thread`
-   * :ref:`ug_zigbee`
+Coexistence based on MPSL CX API
+================================
 
-.. _ug_radio_cx_requirements:
+It is based on :ref:`nrfxlib:mpsl_cx` provided by the :ref:`nrfxlib:mpsl` (MPSL) library and is recommended for new designs.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:35]:
```suggestion
This coexistence feature is based on :ref:`nrfxlib:mpsl_cx` provided by the :ref:`nrfxlib:mpsl` (MPSL) library.
It is recommended for new designs.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 8]
```diff
@@ -1,103 +1,117 @@
...
 
-Enabling MPSL
--------------
+The following SR protocols are compatible with the coexistence based on MPSL CX API:
 
-The MPSL library provides an API to handle signals coming from the PTA and to control requests sent to PTA.
-See the :ref:`nrfxlib:mpsl_lib` in the nrfxlib documentation for details.
+   * :ref:`Bluetooth LE <ug_ble_controller>` if implementation used is :ref:`SoftDevice Controller <nrfxlib:softdevice_controller>`
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:39]:
```suggestion
   * :ref:`Bluetooth LE <ug_ble_controller>`, if the implementation used is the :ref:`SoftDevice Controller <nrfxlib:softdevice_controller>`.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 9]
```diff
@@ -1,103 +1,117 @@
...
+   * Protocols based on :ref:`nrfxlib:nrf_802154` such as :ref:`ug_thread` and :ref:`ug_zigbee`.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:40]:
```suggestion
   * Protocols based on :ref:`nrfxlib:nrf_802154`, such as :ref:`ug_thread` and :ref:`ug_zigbee`.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 10]
```diff
@@ -1,103 +1,117 @@
...
 
-.. note::
-   When using the nRF5340, add the :kconfig:option:`CONFIG_MPSL` option only for the NET core.
+   * :ref:`ug_radio_mpsl_cx_nrf700x`
+   * :ref:`ug_radio_mpsl_cx_generic_3wire`
+   * :ref:`ug_radio_mpsl_cx_custom`
 
-Enabling Wi-Fi Coexistence
---------------------------
+The |NCS| provides a wrapper that configures Wi-Fi Coexistence based on devicetree (DTS) and Kconfig information.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:48]:
```suggestion
The |NCS| provides a wrapper that configures Wi-Fi Coexistence based on devicetree source (DTS) and Kconfig information.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 11]
```diff
@@ -1,103 +1,117 @@
...
 
-To enable Wi-Fi Coexistence, do as follows:
+Common requirements to use coexistence based on MPSL CX API:
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:50]:
```suggestion
The following are the common requirements to use coexistence based on the MPSL CX API:
```

**Review by FrancescoSer — CHANGES_REQUESTED** [review]
Just some suggestions

File: doc/nrf/app_dev/wifi_coex/index.rst (hunk 2 above)
Comment (jciupis) [reply doc/nrf/app_dev/wifi_coex/index.rst:55]: Yes, it's very good now 👍

**Review by jciupis — APPROVED** [review]

File: doc/nrf/protocols.rst [hunk 12]
```diff
@@ -18,5 +18,7 @@ They introduce you to concepts that are important to work with the protocol and
    protocols/multiprotocol/index
    protocols/nfc/index
    protocols/thread/index
-   protocols/wifi/index
    protocols/zigbee/index
+   protocols/wifi/index
+
+For Wi-Fi coexistence with short range protocols (Bluetooth LE, Thread etc.), see the :ref:`ug_radio_coex` page.
```
Comment (FrancescoSer) [inline doc/nrf/protocols.rst:24]:
```suggestion
For details on Wi-Fi coexistence with short range protocols (like Bluetooth LE or Thread), see the :ref:`ug_radio_coex` page.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 13]
```diff
@@ -1,103 +1,117 @@
...
+
+1. Make sure that Kconfig option :kconfig:option:`CONFIG_MPSL` is enabled.
+   Please note that some protocol drivers like :ref:`SoftDevice Controller <nrfxlib:softdevice_controller>` enable this option by default.
+2. Make sure that Kconfig option :kconfig:option:`CONFIG_MPSL_CX` is enabled.
+3. Make sure you meet requirements specific to selected coexistence implementation.
+   These include at least selecting one of supported values of :kconfig:option:`CONFIG_MPSL_CX_CHOICE` choice.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:56]:
```suggestion
   These include at least selecting one of supported values of the :kconfig:option:`CONFIG_MPSL_CX_CHOICE` choice.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 14]
```diff
@@ -1,103 +1,117 @@
...
+   For details on requirements specific to selected coexistence implementation please refer to the documentation of that implementation.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:57]:
```suggestion
   For details on the requirements specific to the selected coexistence implementation, consult its documentation.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 15]
```diff
@@ -1,103 +1,117 @@
...
+4. Make sure a ``nrf_radio_coex`` node appropriate for selected implementation is present in the devicetree.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:58]:
```suggestion
4. Ensure that the configuration of the ``nrf_radio_coex`` node appropriate for the selected implementation is present in the devicetree.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 16]
```diff
@@ -1,103 +1,117 @@
...
+   When using one of the supported implementations, you must use the ``nrf_radio_coex`` name for the node.
+   However, if you add a custom user implementation, you can also use a different name.
+   Please note that some boards supported by the |NCS| (like :ref:`nrf7002dk_nrf5340 <nrf7002dk_nrf5340>`) provide this node out of the box.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:61]:
```suggestion
   Some boards supported by the |NCS| (like :ref:`nrf7002dk_nrf5340 <nrf7002dk_nrf5340>`) provide this node by default.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 17]
```diff
@@ -1,103 +1,117 @@
...
+   You can provide the node using the devicetree source file of the target board or an overlay file.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:62]:
```suggestion
   You can provide the node using either the devicetree source file of the target board or an overlay file.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 18]
```diff
@@ -1,103 +1,117 @@
...
+   See :ref:`zephyr:dt-guide` for more information about the DTS data structure, and :ref:`zephyr:dt_vs_kconfig` for information about differences between DTS and Kconfig.
+5. For the nRF5340 SoC make sure the GPIO pins required for communication with PTA are handed over to the network core.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:64]:
```suggestion
5. On the nRF5340 SoC, the GPIO pins required for the communication with the PTA must be handed over to the network core.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 19]
```diff
@@ -1,103 +1,117 @@
...
+
+Please refer to :ref:`ug_radio_coex_mpsl_cx_based` for general requirements of this implementation.
+
+Hardware description
+--------------------
 
-.. figure:: images/coex_generic_3pin_pta.svg
-   :alt: PTA interface supported by the ``nrf_radio_coex`` node with an nRF52 Series SoC
+The nRF700x interface consists of signals listed in the table below.
+The *Pin* is the pin name of nRF700x device.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:98]:
```suggestion
The *Pin* is the pin name of the nRF700x device.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 20]
```diff
@@ -135,12 +149,41 @@ To configure the hardware for a PTA inside the nRF700x series chips:
...
+   Pin           Direction  Description                        DT property
+   ============  =========  =================================  ==============
+   REQUEST       Out        Request signal                     req-gpios
+   PRIORITY      Out        SR transaction direction TX or RX  pri-dir-gpios
+   GRANT         In         Grant signal                       grant-gpios
+   ============  =========  =================================  ==============
+
+
+Enabling generic three wire coexistence
+---------------------------------------
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:187]:
```suggestion

To enable the generic three-wire coexistence, do the following:


```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 21]
```diff
@@ -135,12 +149,41 @@ To configure the hardware for a PTA inside the nRF700x series chips:
...
+   ============  =========  =================================  ==============
+   Pin           Direction  Description                        DT property
+   ============  =========  =================================  ==============
+   REQUEST       Out        Request signal                     req-gpios
+   PRIORITY      Out        SR transaction direction TX or RX  pri-dir-gpios
+   GRANT         In         Grant signal                       grant-gpios
+   ============  =========  =================================  ==============
+
+
+Enabling generic three wire coexistence
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:185]:
```suggestion
Enabling generic three-wire coexistence
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 22]
```diff
@@ -135,12 +149,41 @@ To configure the hardware for a PTA inside the nRF700x series chips:
...
+---------------------------------------
 
 1. Add the following node in the devicetree source file:
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:188]:
```suggestion
1. Add the following node to the devicetree source file:
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 23]
```diff
@@ -178,79 +221,58 @@ To configure the hardware for a PTA using a 3-wire interface:
...
-.. _ug_radio_cx_wifi_coex_impls:
-
-Wi-Fi Coexistence implementations
-*********************************
-
-The following CX implementations are available:
-
-* :ref:`ug_radio_cx_nrf700x_impl`
-* :ref:`ug_radio_cx_thread_impl`
+#. Make sure the following Kconfig options are enabled:
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:224]:
```suggestion
#. Enable the following Kconfig options:
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 24]
```diff
@@ -178,79 +221,58 @@ To configure the hardware for a PTA using a 3-wire interface:
...
-  It contains information about the type of operation (either RX or TX) to perform.
-  It can be configured by setting the ``status0-gpios`` property of the ``nrf_radio_coex`` devicetree node.
-* GRANT pin - It is the input of the SoC controlled by the PTA.
-  It asserts when PTA grants access to the RF to the 802.15.4 and deasserts when it denies the access.
-  It can be configured by setting the ``grant-gpios`` property of the ``nrf_radio_coex`` devicetree node.
+User custom coexistence implementations
+=======================================
 
-The support for this interface is provided in a single-file plugin located in the sdk-nrf repo, in the :file:`subsys/mpsl/cx` directory.
+Please refer to :ref:`ug_radio_coex_mpsl_cx_based` for general requirements that must be fulfilled by the implementation.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:235]:
```suggestion
Refer to :ref:`ug_radio_coex_mpsl_cx_based` for the general requirements that must be fulfilled by the implementation.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 25]
```diff
@@ -178,79 +221,58 @@ To configure the hardware for a PTA using a 3-wire interface:
...
-* REQUEST pin - It is the output controlled by the protocol driver.
-  It is asserted to request RF access from the PTA.
-  It can be configured by setting the ``req-gpios`` property of the ``nrf_radio_coex`` devicetree node.
-* PRIORITY pin - It is the output controlled by the protocol driver.
-  It contains information about the type of operation (either RX or TX) to perform.
-  It can be configured by setting the ``status0-gpios`` property of the ``nrf_radio_coex`` devicetree node.
-* GRANT pin - It is the input of the SoC controlled by the PTA.
-  It asserts when PTA grants access to the RF to the 802.15.4 and deasserts when it denies the access.
-  It can be configured by setting the ``grant-gpios`` property of the ``nrf_radio_coex`` devicetree node.
+User custom coexistence implementations
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:232]:
```suggestion
Custom coexistence implementations
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 26]
```diff
@@ -178,79 +221,58 @@ To configure the hardware for a PTA using a 3-wire interface:
...
+=======================================
 
-The support for this interface is provided in a single-file plugin located in the sdk-nrf repo, in the :file:`subsys/mpsl/cx` directory.
+Please refer to :ref:`ug_radio_coex_mpsl_cx_based` for general requirements that must be fulfilled by the implementation.
 
-Adding support for the CX nRF700x interface
--------------------------------------------
+To add custom coexistence implementation based on MPSL CX API complete following steps:
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:237]:
```suggestion
To add a custom coexistence implementation based on the MPSL CX API, complete following steps:
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 27]
```diff
@@ -178,79 +221,58 @@ To configure the hardware for a PTA using a 3-wire interface:
...
 
-The nRF700x interface is supported out of the box.
-To use it, complete the following steps:
+1. Determine hardware interface for your PTA.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:239]:
```suggestion
1. Determine the hardware interface of your PTA.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 28]
```diff
@@ -178,79 +221,58 @@ To configure the hardware for a PTA using a 3-wire interface:
...
+   If your PTA uses interface other than already provided by |NCS| you need to provide devicetree binding file.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:240]:
```suggestion
   If your PTA uses an interface different from the ones already provided by the |NCS|, you need to provide a devicetree binding file.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 29]
```diff
@@ -178,79 +221,58 @@ To configure the hardware for a PTA using a 3-wire interface:
...
+   Please see :file:`nrf/dts/bindings/radio_coex/generic-radio-coex-three-wire.yaml` file as an example.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:241]:
```suggestion
   See :file:`nrf/dts/bindings/radio_coex/generic-radio-coex-three-wire.yaml` file for an example.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 30]
```diff
@@ -178,79 +221,58 @@ To configure the hardware for a PTA using a 3-wire interface:
...
+#. Extend Kconfig choice :kconfig:option:`CONFIG_MPSL_CX_CHOICE` with Kconfig option allowing to select the new coex implementation.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:242]:
```suggestion
#. Extend the Kconfig choice :kconfig:option:`CONFIG_MPSL_CX_CHOICE` with a Kconfig option allowing to select the new coex implementation.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 31]
```diff
@@ -178,79 +221,58 @@ To configure the hardware for a PTA using a 3-wire interface:
...
+#. Write implementation for your PTA.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:243]:
```suggestion
#. Write the implementation for your PTA.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 32]
```diff
@@ -178,79 +221,58 @@ To configure the hardware for a PTA using a 3-wire interface:
...
+   Please see :file:`nrf/subsys/mpsl/cx/thread/mpsl_cx_thread.c` file as an example.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:244]:
```suggestion
   See the :file:`nrf/subsys/mpsl/cx/thread/mpsl_cx_thread.c` file for an example.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 33]
```diff
@@ -178,79 +221,58 @@ To configure the hardware for a PTA using a 3-wire interface:
...
+   Add C source file(s) with implementation, which must contain the following parts:
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:245]:
```suggestion
   Add the C source file(s) with the implementation, which must contain the following parts:
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 34]
```diff
@@ -178,79 +221,58 @@ To configure the hardware for a PTA using a 3-wire interface:
...
 
-1. Add the devicetree `nrf_radio_coex` node as described in :ref:`ug_radio_cx_nrf700x_hw_if`.
-#. Set the :kconfig:option:`CONFIG_MPSL_CX` Kconfig option to ``y``.
-   On the nRF5340, you must apply the Kconfig option also to the network core.
+   * Implementation of functions required by interface structure :c:struct:`mpsl_cx_interface_t`.
+     Please refer to :ref:`MPSL CX API <mpsl_api_sr_cx>` for details.
+   * Initialization code initializing required hardware resources, based on devicetree information.
+   * Call to function :c:func:`mpsl_cx_interface_set()` on system initialization.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:250]:
```suggestion
   * The implementation of the functions required by the interface structure :c:struct:`mpsl_cx_interface_t`.
     Refer to :ref:`MPSL CX API <mpsl_api_sr_cx>` for details.
   * The initialization code initializing the required hardware resources, based on devicetree information.
   * A call to the function :c:func:`mpsl_cx_interface_set()` during the system initialization.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 35]
```diff
@@ -178,79 +221,58 @@ To configure the hardware for a PTA using a 3-wire interface:
...
 
-.. _ug_radio_cx_thread_impl:
+#. Add necessary CMakeLists.txt entries to get your code compiled when the new Kconfig choice option you added is selected.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:252]:
```suggestion
#. Add the necessary CMakeLists.txt entries to get your code compiled when the new Kconfig choice option you added is selected.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 36]
```diff
@@ -178,79 +221,58 @@ To configure the hardware for a PTA using a 3-wire interface:
...
-* REQUEST pin - it is the output controlled by the protocol driver.
-  It is asserted to request RF access from the PTA.
-  It can be configured by setting the ``req-gpios`` property of the ``nrf_radio_coex`` devicetree node.
-* PRIORITY pin - it is the output controlled by the protocol driver.
-  It is asserted if the operation priority is high and deasserted if it is low.
-  It can be configured by setting the ``pri-dir-gpios`` property of the ``nrf_radio_coex`` devicetree node.
-* GRANT pin - it is the input of the SoC controlled by the PTA.
-  It asserts when PTA grants access to the RF to the 802.15.4 and deasserts when it denies the access.
-  It can be configured by setting the ``grant-gpios`` property of the ``nrf_radio_coex`` devicetree node.
+Please refer to :ref:`ug_radio_coex_bluetooth_only_based` for general requirements of this implementation.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:259]:
```suggestion
Refer to :ref:`ug_radio_coex_bluetooth_only_based` for the general requirements of this implementation.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 37]
```diff
@@ -178,79 +221,58 @@ To configure the hardware for a PTA using a 3-wire interface:
...
 
-The support for this interface is provided in a single-file plugin located in the sdk-nrf repo, in the :file:`subsys/mpsl/cx` directory.
+The Bluetooth-only 1-wire coexistence feature allows the :ref:`SoftDevice Controller <nrfxlib:softdevice_controller>` to coexist alongside an LTE device on a separate chip.
+It is specifically designed for the coex interface of nRF9160.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:262]:
```suggestion
It is specifically designed for the coex interface of the nRF9160.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 38]
```diff
@@ -178,79 +221,58 @@ To configure the hardware for a PTA using a 3-wire interface:
...
+The implementation is based on :ref:`nrfxlib:mpsl_bluetooth_coex_1wire` which is provided into the :ref:`nrfxlib:mpsl` (MPSL) library.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:263]:
```suggestion
The implementation is based on :ref:`nrfxlib:mpsl_bluetooth_coex_1wire`, which is provided into the :ref:`nrfxlib:mpsl` (MPSL) library.
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 39]
```diff
@@ -178,79 +221,58 @@ To configure the hardware for a PTA using a 3-wire interface:
...
+Make sure that the following Kconfig options are enabled:
 
-Custom user implementations
-===========================
+   * :kconfig:option:`CONFIG_MPSL_CX`
+   * :kconfig:option:`CONFIG_MPSL_CX_BT_1WIRE`
 
-Implementing a custom user CX implementation is described in :ref:`nrfxlib:mpsl_cx`.
-If the ``nrf_radio_coex`` devicetree node is not sufficient for the implementation, you must add a custom devicetree node.
+Configuration is set using the devicetree (DTS).
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:274]:
```suggestion
The configuration is set using devicetree (DTS).
```

File: doc/nrf/app_dev/wifi_coex/index.rst [hunk 40]
```diff
@@ -178,79 +221,58 @@ To configure the hardware for a PTA using a 3-wire interface:
...
+For more information about devicetree overlays, see :ref:`zephyr:use-dt-overlays`.
+A sample devicetree overlay is available at :file:`samples/bluetooth/radio_coex_1wire/boards/nrf52840dk_nrf52840.overlay`.
+The elements are described in the bindings: :file:`dts/bindings/radio_coex/sdc-radio-coex-one-wire.yaml`.
+A sample application for 1-wire coexistence can be found at :file:`samples/bluetooth/radio_coex_1wire`.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/wifi_coex/index.rst:278]:
```suggestion
See :file:`samples/bluetooth/radio_coex_1wire` for a sample application using 1-wire coexistence.
```

**Review by FrancescoSer — CHANGES_REQUESTED** [review]

**Review by ahasztag — APPROVED** [review]

**Comment by ankuns** [comment]
@FrancescoSer, I have addressed all your suggestions.

**Review by FrancescoSer — APPROVED** [review]
